openeo/extra/job_management/thread_worker.py

```diff
@@ -0,0 +1,183 @@
+import concurrent.futures
+import logging
+from typing import Any, Dict, List, Tuple
+
+import openeo
+
+_log = logging.getLogger(__name__)
+
+
+class _JobManagerWorkerThreadPool:
+    """
+    A worker thread pool for processing job management tasks asynchronously.
+
+    This thread pool is designed to handle tasks such as starting jobs on a backend.
+    It uses a `ThreadPoolExecutor` to process tasks concurrently and tracks their results
+    using futures.
+
+    Attributes:
+        Public:
+            WORK_TYPE_START_JOB (str): A constant representing the work type (e.g. starting jobs).
+        Private:
+            _executor (concurrent.futures.ThreadPoolExecutor): The thread pool executor used to manage worker threads.
+            _futures (List[concurrent.futures.Future, str]): A list of futures.
+    """
+
+    WORK_TYPE_START_JOB = "start_job"
+
+    def __init__(self, max_workers: int = 2):
+        """
+        Initialize the worker thread pool.
+
+        Args:
+            max_workers (int): The maximum number of worker threads to use in the thread pool.
+                              Defaults to 2.
+        """
+        self._executor = concurrent.futures.ThreadPoolExecutor(max_workers=max_workers)
+        self._futures: List[concurrent.futures.Future] = []
+
+    def submit_work(self, work_type: str, work_args: Tuple[Any, ...]) -> None:
+        """
+        Submit a work item to the thread pool for processing.
+
+        Args:
+            work_type (str): The type of work to be performed.
+            work_args (Tuple[Any, ...]): A tuple of arguments required to process the work item.
+        """
+        # Validate work type and arguments
+        self._validate_work_args(work_type, work_args)
```

Comment:
With adding things like validation, the logic around a works item will get more and more scattered in various places. I think it makes sense to encapsulate all task aspects in a dedicated class, to replace the ad-hoc tuples. This class can then even contain the actual implementation of the work (e.g. starting a job in this case), which allows to decouple the thread pool aspects from the specifics of a task.

e.g. very roughly (using dataclasses to compactly define a "named tuple"):

```
import dataclasses

@dataclasses.dataclass(frozen=True)
class JobStartTask:
    root_url: str
    bearer_token: str
    job_id: str

    # possibly an __init__ here, e.g. to do validation if that is necessary

   def execute(self):
       # create connection and start job ...
```